src/http/response.rs

```diff
@@ -0,0 +1,152 @@
+use std::fmt;
+
+use header::{Header, Headers};
+use http::{MessageHead, ResponseHead, Body, RawStatus};
+use status::StatusCode;
+use version::HttpVersion;
+
+/// A response for a client request to a remote server.
+pub struct Response<B = Body> {
+    version: HttpVersion,
+    headers: Headers,
+    status: StatusCode,
+    raw_status: RawStatus,
+    body: Option<B>,
+}
+
+impl<B> Response<B> {
+    /// Constructs a default response
+    #[inline]
+    pub fn new() -> Response<B> {
+        Response::default()
+    }
+
+    /// Get the HTTP version of this response.
+    #[inline]
+    pub fn version(&self) -> &HttpVersion { &self.version }
```

Comment:
Returning a `&HttpVersion` instead of `HttpVersion` might have been inconsistent on one of the types. The intent was to just always return it by value, since it's smaller than a pointer and `Copy`. Could you update that here?